### PR TITLE
feat: MS Teams Mentions

### LINKED ISSUE #1: Improve MS Teams support
Would it be possible to add support for MS teams as well?

----------

edit by @DavideViolante:
- [x] MS Teams works
- [x] message with links
- [ ] mentions

### PR DESCRIPTION
**Pull request by arranfw**
## Description
- Adds support for MS teams mentions, resolves #1 

Follows Microsoft's example [here](https://github.com/DavideViolante/pr-reviews-reminder-action/pull/68)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,7 +22,9 @@ The channel name, eg: `#general` (optional).
 
 ### github-provider-map
 
-A string like this `"githubusername1:provideruserid1,githubusername2:provideruserid2,..."` to define the mapping between GitHub usernames and Slack user IDs (optional). Example: `"DavideViolante:UEABCDEFG,foobar:UAABCDEFG"`. Note: it's case sensitive! [How to find Slack User IDs](https://www.google.com/search?q=find+slack+user+id).
+A string like this `"githubusername1:provideruserid1,githubusername2:provideruserid2,..."` to define the mapping between GitHub usernames and Slack/MS Teams user IDs (optional). Example: `"DavideViolante:UEABCDEFG,foobar:UAABCDEFG"`. Note: it's case sensitive! [How to find Slack User IDs](https://www.google.com/search?q=find+slack+user+id).
+
+For MS Teams, the provider user ID can be an [MS teams user ID or a UPN](https://github.com/arranfw/pr-reviews-reminder-action/releases) Example: `"DavideViolante:davidviolant@onmicrosoft.com,foobar:foobar@onmicrosoft.com"`.
```
Comment (DavideViolante) [inline README.md:27]: This link should show a page explaining how to get a MS Teams member ID or UPN, but I believe it's wrong (it's linking your releases at the moment).

File: functions.js [hunk 2]
```diff
@@ -61,25 +61,104 @@ function stringToObject(str) {
...
+  const github2providerEntries = Object.entries(github2provider);
+  const mentionObjects = github2providerEntries.map(([githubId, providerId]) => ({
+    type: `mention`,
+    text: `<at>${githubId} UPN</at>`,
+    mentioned: {
+      id: providerId,
+      name: githubId,
+    },
+  })).filter((mention) => pr2user.find((item) => item.login === mention.mentioned.name));
+  // Filter for users who have been requested in a review
```
Comment (DavideViolante) [inline functions.js:101]: Can you split this into 2 variables please? Example:
const mapped = example.map(...)
const filtered = mapped.filter(...)

File: functions.js [hunk 3]
```diff
@@ -61,25 +61,104 @@ function stringToObject(str) {
...
+
+  return mentionObjects;
+}
+
+/**
+ * Formats channel and slack message text into a request object
+ * @param {*} channel channel to send the message to
+ * @param {*} message slack message text
```
Comment (DavideViolante) [inline functions.js:109]:
```suggestion
 * @param {String} channel channel to send the message to
 * @param {String} message slack message text
```

**Review by DavideViolante — CHANGES_REQUESTED** [review]
It seems a very nice PR, thank you! I left you a few things to review.

File: README.md (hunk 1 above)
Comment (arranfw) [reply README.md:27]: Haha oops!

**Pull request merged**